rfc/eol_kubernetes_v1.md

```diff
@@ -0,0 +1,109 @@
+ I am also likely to be confused by an apparent mismatch between Spinnaker documentation and what I see in the UI, due to uncertainty around which Kubernetes provider I have actually installed.
+- **Focus Spinnaker/Kubernetes community on active areas of development**.
+The Spinnaker Kubernetes SIG and Slack channels operate best with a common foundation from which to work, one that reflects the opinions and best practices of our community.
+- **The user experience for existing V1 users is poor.**
+  Existing V1 users receive almost no support or up-to-date content from the Spinnaker community.
+  Additionally, they encounter bugs and performance problems that are unlikely to be resolved.
+  We have made significant user experience enhancements and performance improvements in the V2 provider, and will continue to invest solely in V2 for Kubernetes users.
+- **Reduce the maintenance burden for Spinnaker developers.**
+  Currently, the V1 provider is in maintenance mode: although we are no longer adding or accepting enhancements, we are still responsible for triaging and fixing bugs and upgrading dependencies.
+  This is a non-trivial amount of overhead that subtracts from the time we can commit to improving the V2 provider.
+
+## Timeline
+
+Each set of milestones should be complete by the time the corresponding Spinnaker version is released.
+
+| Spinnaker Version | Release Window Opens | Milestones |
+|-------------------|----------------------|------------|
+| 1.18              | 2020-01-06           | Gain consensus on this RFC. Highlight in 1.18 curated changelog that V1 will be removed in 1.21.
+| 1.19              | 2020-03-02           | Publish a post to the Spinnaker Community Blog explaining why we are removing V1 and how to migrate to V2. Add a feature flag that must be enabled before deploying to V1 accounts. Add appropriate warnings to Halyard, Clouddriver, and Deck. Highlight in 1.19 curated changelog that V1 will be removed in 1.21.
+| 1.20              | 2020-04-27           | Complete audit of community docs and blog posts for V1-related content to update. Highlight in 1.20 curated changelog that V1 will be removed in 1.21.
+| 1.21              | 2020-06-22           | Remove V1-only code from all Spinnaker microservices.
+
+## Design
+
+- [x] Determine if there are user stories addressed by the V1 provider but not the V2 provider that we should consider adding to V2 before removing V1.
+  - [x] Identify and interview existing V1 users in order to learn why they are still using the V1 provider (Thank you to everyone who took the time to speak with us!).
+  - [x] Identify and interview people currently evaluating Spinnaker in order to learn if the level of abstraction exposed by V1’s wizards would have value as an onboarding tool in V2
+  (Short answer: most users think that manifest composition is now out of scope for Spinnaker because there is now such a rich ecosystem of Kubernetes-native tooling for manifest templating).
```

Comment:
Question about this - does this mean that users don't want something like `Bake (Manifest)` as a pipeline stage anymore? Or that they don't see a Spinnaker-specific solution as necessary?

Reply:
Definitely the second: most don't see a Spinnaker-specific solution as necessary. Spinnaker should integrate well with, but not attempt to replicate the core feature set of, tools like Kustomize and Helm. Because operators want to use these Kubernetes-native templating tools to provide some level of abstraction to their end-users over direct manifest composition, Spinnaker should probably not provide its own abstraction (in the case of V1, the UI wizard). I think the Bake (Manifest) stage is still super useful for the many users who want Helm or Kustomize manifest hydration to happen in Spinnaker rather than CI. Does this match your understanding?

Reply:
> Does this match your understanding?

Yes, totally. Having the Bake stage is useful for new users as they onboard because it's one less question to answer. I agree that building something Spinnaker specific is out of scope for what we want to do.